bundles/org.eclipse.passage.lbc.base/src/org/eclipse/passage/lbc/internal/base/persistence/LockFile.java

```diff
@@ -0,0 +1,57 @@
+/*******************************************************************************
+ * Copyright (c) 2020 ArSysOp
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * https://www.eclipse.org/legal/epl-2.0/.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     ArSysOp - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.passage.lbc.internal.base.persistence;
+
+import java.nio.file.Path;
+import java.util.Objects;
+import java.util.function.Supplier;
+
+import org.eclipse.passage.lbc.internal.api.persistence.BoundLicense;
+import org.eclipse.passage.lic.internal.api.conditions.Condition;
+
+/**
+ * @since 1.0
+ */
+public final class LockFile implements Supplier<Path> {
+
+	private final Supplier<Path> folder;
+	private final String identifier;
+
+	public LockFile(Supplier<Path> folder, BoundLicense license) {
+		Objects.requireNonNull(folder, "LockFile::folder"); //$NON-NLS-1$
+		Objects.requireNonNull(license, "LockFile::license"); //$NON-NLS-1$
+		this.folder = folder;
+		this.identifier = license.identifier().get().get();
+	}
+
+	public LockFile(Supplier<Path> folder, Condition license) {
+		Objects.requireNonNull(folder, "LockFile::folder"); //$NON-NLS-1$
+		Objects.requireNonNull(license, "LockFile::license"); //$NON-NLS-1$
+		this.folder = folder;
+		this.identifier = license.identifier();
+	}
```

Comment:
Both these constructors do the same work. It should be done ones by a single primary constructor (from `Supplier<Path>` and `String`), most propapbly `private`. These two guys should just appeal to it. 

You'll lose ability to demand not-nullability for `BoundLicense` and `Condition` directly, but will get NPE for violators eitherway. Which is perfectly fine, especially having a javadoc-ed contract.